src/GafferScene/TweakPlug.cpp

```diff
@@ -225,7 +228,43 @@ class NumericTweak
 	}
 
 	template<typename T>
-	void operator()( T * data, typename std::enable_if<SupportsArithData<T>::value>::type *enabler = nullptr ) const
+	void operator()( T * data, typename std::enable_if<TypeTraits::IsNumericSimpleTypedData<T>::value>::type *enabler = nullptr ) const
+	{
+		const T *sourceDataCast = runTimeCast<const T>( m_sourceData );
+		const T *tweakDataCast = runTimeCast<const T>( m_tweakData );
+		switch( m_mode )
+		{
+			case TweakPlug::Add :
+				data->writable() = sourceDataCast->readable() + tweakDataCast->readable();
+				break;
+			case TweakPlug::Subtract :
+				data->writable() = sourceDataCast->readable() - tweakDataCast->readable();
+				break;
+			case TweakPlug::Multiply :
+				data->writable() = sourceDataCast->readable() * tweakDataCast->readable();
+				break;
```

Comment:
I don't think we should need to duplicate this logic between the two operators, or even need to introduce a new operator. I think we could just define our own `min` function and use that :

```
template<typename T>
T myMin( ... )
{
    return std::min( ... );
}

template<typename T>
Imath::Vec<T> myMin( ... )
{
    // Do stuff with dimensions
}
```

Reply:
I maybe don't have the best solution in 28eadea4221e3faf6fe4d8ffad2b0591d78fc7fe, but it's a start. Is there a way to handle multiple vector types with templates? It seems to be forcing me to handle `Vec2`, `Vec3`, `Color3`, etc. independently.

Reply:
> I maybe don't have the best solution in https://github.com/GafferHQ/gaffer/commit/28eadea4221e3faf6fe4d8ffad2b0591d78fc7fe, but it's a start.

Yep, I definitely think this is an improvement over the duplication of the higher level logic.

> Is there a way to handle multiple vector types with templates? 

Here's one approach, using `if constexpr` to check `IECore::TypeTraits::IsVec` : https://github.com/johnhaddon/gaffer/commit/420ee0a05796cf60a113f1b0ca88f5ca36ecc547.

Reply:
Much better, thanks for describing that. Fixed in 4680e425b1244a8d2274827277409aa49961dcc0